paasta_tools/api/views/instance.py

```diff
@@ -336,37 +356,86 @@ def marathon_smartstack_status(
         for attribute_value, slaves in grouped_slaves.items()
     }
 
-    expected_smartstack_count = marathon_tools.get_expected_instance_count_for_namespace(
+    expected_instance_count = marathon_tools.get_expected_instance_count_for_namespace(
         service, instance, settings.cluster
     )
     expected_count_per_location = int(
-        expected_smartstack_count / len(slave_hostname_by_location)
+        expected_instance_count / len(slave_hostname_by_location)
     )
-    smartstack_status: MutableMapping[str, Any] = {
+    service_mesh_status: MutableMapping[str, Any] = {
         "registration": registration,
         "expected_backends_per_location": expected_count_per_location,
         "locations": [],
     }
 
     for location, hosts in slave_hostname_by_location.items():
-        synapse_host = hosts[0]
-        sorted_backends = sorted(
-            get_backends(
+        if service_mesh == ServiceMesh.SMARTSTACK:
```

Comment:
it really feels like the smartstack and envoy parts should just be sparate methods.  Is that where you were running into mypy issues?  If so please try a little more, and maybe I can take a crack at it if you can't make it work.